azurelinuxagent/common/protocol/wire.py

```diff
@@ -768,6 +799,35 @@ def _update_from_goal_state(self, refresh_type):
 
         raise ProtocolError("Exceeded max retry updating goal state")
 
+    def _save_goal_state_if_necessary(self, new_goal_state, refresh_type):
```

Comment:
the name of this function is confusing, I think, especially given its similarity with the existing _save_goal_state. This new function is also updating the current goal state in the wire client (on top of saving to disk). let's rename it

Reply:
Renamed to _cache_goal_state_and_update_host_plugin